File: 프로그래머스/0/120956. 옹알이 （1）/옹알이 （1）.cpp
#include <iostream>
#include <vector>
#include <string>
#include <algorithm>

using namespace std;

int solution(vector<string> babbling) {
    int answer = 0;
    vector<string> can_word = {"aya", "ye", "woo", "ma"};
    
    for (string b : babbling) {
        string temp = b;
        
        for (string w : can_word) {
            size_t pos = temp.find(w);
            if (pos != string::npos) {
                temp.replace(pos, w.length(), " ");
            }
        }
        
        temp.erase(remove(temp.begin(), temp.end(), ' '), temp.end());
        
        if (temp.empty()) {
            answer++;
        }
    }
    
    return answer;
}

File: 프로그래머스/0/120956. 옹알이 （1）/옹알이 （1）.js
function solution(babbling) {
    var answer = 0;
    const can_word = ["aya", "ye", "woo", "ma"];
    
    for (let b of babbling) {
        
        const word_set = new Set();
        
        for(let w of can_word) {
            while(b.includes(w)) {
                if(!word_set.has(w)){
                    word_set.add(w);
                    b = b.replace(w, ' ');
                }
                else
                    break;
            }
        }
        
        if(b.trim().length === 0)
            answer++;
    }
    
    return answer;
}